Upgrade geth to 1.10.4

cammellos opened the pull request:
This commit upgrades geth to 1.10.4, in order to get ready for the London fork.

Because of the many changes in geth, there are quite a few changes required:

- Services have changed interface, and are now Lifecycles
- I have moved all the services in StatusNode, as before they were scattered around between status-node,geth_backend and geth_node
- Upgraded a bunch of libraries (go-ens, iconv)
- Removed some unused ens contract code, only used in testing
- Removed mailserver registry code, not used anymore
- Fixed some upgrade issues
- Upgraded to the new london signer
- Added a populate db script from a different PR, useful for testing
- Removed e2e tests, as most of them seems to be testing node usage outside of mobile/desktop


Because of the time constraints I had to cut some corners and remove some of the tests.

A lot of it was testing blockchain synchronization, which we don't use, or using other functionality (status api), that we never used.

rasom committed:
Add some eip 1559 related rpc methods to router

cammellos committed:
Upgrade go-ens

Go ens needs to be updated to be compatible with the lastest geth
version

cammellos committed:
Memove mailserver registry and update ierc20 contract bindings

This commit removes the mailserver contract registry, as not used
anymore, and upgrade the ERC20 contract bindings.

cammellos committed:
Fix hash & start moving to lifecycle

Fix typeddata has and start moving from the legacy geth services
interfaces to the new Lifecycle interface

cammellos committed:
Fix wallet big int conversion

cammellos committed:
Fix iconcache

Iconcache library needed to be upgrade to be compatible with the
libraries required in the latest geth.

cammellos committed:
Upgrade geth

Point to the new fork of geth.

cammellos committed:
Move services to status-node

Move all the services to status-node, as some of them were there, some
of them were in the geth backend and scattered around.

cammellos committed:
Add populate db script

This commit adds a populate db script, used by QA, but also very useful
to test Login/Logout flows.

cammellos committed:
Upgrade to london signer and cleanup

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?
- [ ] Have you tested changes with mobile or desktop version?

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (60)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | f092b818 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/1/) | 2021-07-09 09:17:07 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210709-091331-f092b8.zip) |
| :heavy_check_mark: | f092b818 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/1/) | 2021-07-09 09:17:09 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210709-091331-f092b8.zip) |
| :heavy_check_mark: | f092b818 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/1/) | 2021-07-09 09:21:00 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210709-091331-f092b8.aar) |
| | | | | | | |
| :x: | 3e81fe27 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/2/) | 2021-07-09 13:22:18 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/2/consoleText) |
| :x: | 3e81fe27 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/2/) | 2021-07-09 13:22:31 | ~1 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/2/consoleText) |
| :x: | 3e81fe27 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/2/) | 2021-07-09 13:22:45 | ~1 min | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/2/consoleText) |
| | | | | | | |
| :x: | 3cbe7f04 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/3/) | 2021-07-09 13:28:24 | ~30 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/3/consoleText) |
| :x: | 3cbe7f04 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/3/) | 2021-07-09 13:28:56 | ~1 min | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/3/consoleText) |
| :x: | 3cbe7f04 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/3/) | 2021-07-09 13:29:52 | ~1 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/3/consoleText) |
| | | | | | | |
| :x: | 8525d6e7 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/4/) | 2021-07-09 13:43:18 | ~38 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/4/consoleText) |
| :x: | 8525d6e7 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/4/) | 2021-07-09 13:43:41 | ~57 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/4/consoleText) |
| :x: | 8525d6e7 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/4/) | 2021-07-09 13:44:38 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/4/consoleText) |
| | | | | | | |
| :heavy_check_mark: | 73573ba6 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/5/) | 2021-07-09 14:25:39 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210709-142309-73573b.zip) |
| :heavy_check_mark: | 73573ba6 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/5/) | 2021-07-09 14:26:05 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210709-142309-73573b.zip) |
| :heavy_check_mark: | 73573ba6 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/5/) | 2021-07-09 14:34:20 | ~11 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210709-142309-73573b.aar) |
| | | | | | | |
| :heavy_check_mark: | f02db5a2 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/6/) | 2021-07-12 09:46:49 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210712-094455-f02db5.zip) |
| :heavy_check_mark: | f02db5a2 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/6/) | 2021-07-12 09:48:03 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210712-094500-f02db5.zip) |
| :heavy_check_mark: | f02db5a2 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/6/) | 2021-07-12 09:50:11 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210712-094455-f02db5.aar) |
| | | | | | | |
| :heavy_check_mark: | 2ae27cee | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/7/) | 2021-07-12 10:09:10 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210712-100710-2ae27c.zip) |
| :heavy_check_mark: | 2ae27cee | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/7/) | 2021-07-12 10:09:18 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210712-100710-2ae27c.zip) |
| :heavy_check_mark: | 2ae27cee | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/7/) | 2021-07-12 10:11:46 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210712-100705-2ae27c.aar) |
| | | | | | | |
| :x: | b3c7d76f | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/8/) | 2021-07-12 11:19:41 | ~28 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/8/consoleText) |
| :x: | b3c7d76f | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/8/) | 2021-07-12 11:21:51 | ~2 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/8/consoleText) |
| :heavy_check_mark: | b3c7d76f | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/8/) | 2021-07-12 11:22:33 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210712-111917-b3c7d7.zip) |
| | | | | | | |
| :heavy_check_mark: | aee55926 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/9/) | 2021-07-12 11:36:14 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210712-113412-aee559.zip) |
| :heavy_check_mark: | aee55926 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/9/) | 2021-07-12 11:36:24 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210712-113412-aee559.zip) |
| :heavy_check_mark: | aee55926 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/9/) | 2021-07-12 11:41:14 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210712-113412-aee559.aar) |
| | | | | | | |
| :heavy_check_mark: | 6ee995cd | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/10/) | 2021-07-12 14:19:12 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210712-141706-6ee995.zip) |
| :heavy_check_mark: | 6ee995cd | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/10/) | 2021-07-12 14:19:38 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210712-141706-6ee995.zip) |
| :heavy_check_mark: | 6ee995cd | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/10/) | 2021-07-12 14:24:03 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210712-141701-6ee995.aar) |
| | | | | | | |
| :heavy_check_mark: | 8b3c8947 | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/11/) | 2021-07-12 15:06:39 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210712-150437-8b3c89.zip) |
| :heavy_check_mark: | 8b3c8947 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/11/) | 2021-07-12 15:07:15 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210712-150437-8b3c89.zip) |
| :heavy_check_mark: | 8b3c8947 | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/11/) | 2021-07-12 15:09:57 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210712-150437-8b3c89.aar) |
| | | | | | | |
| :heavy_check_mark: | b9d1e0fe | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/12/) | 2021-07-12 15:29:10 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210712-152658-b9d1e0.zip) |
| :heavy_check_mark: | b9d1e0fe | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/12/) | 2021-07-12 15:29:34 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210712-152658-b9d1e0.zip) |
| :heavy_check_mark: | b9d1e0fe | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/12/) | 2021-07-12 15:34:09 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210712-152653-b9d1e0.aar) |
| | | | | | | |
| :heavy_check_mark: | 29ef233f | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/13/) | 2021-07-13 05:50:44 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210713-054834-29ef23.zip) |
| :heavy_check_mark: | 29ef233f | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/13/) | 2021-07-13 05:50:46 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210713-054834-29ef23.zip) |
| :heavy_check_mark: | 29ef233f | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/13/) | 2021-07-13 05:53:38 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210713-054834-29ef23.aar) |
| | | | | | | |
| :heavy_check_mark: | fc2c017a | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/14/) | 2021-07-13 06:50:00 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210713-064800-fc2c01.zip) |
| :heavy_check_mark: | fc2c017a | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/14/) | 2021-07-13 06:50:03 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210713-064800-fc2c01.zip) |
| :heavy_check_mark: | fc2c017a | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/14/) | 2021-07-13 06:54:03 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210713-064800-fc2c01.aar) |
| | | | | | | |
| :heavy_check_mark: | 0ed9ca19 | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/15/) | 2021-07-14 07:52:18 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210714-074948-0ed9ca.zip) |
| :heavy_check_mark: | 0ed9ca19 | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/15/) | 2021-07-14 07:52:19 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210714-074948-0ed9ca.zip) |
| :heavy_check_mark: | 0ed9ca19 | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/15/) | 2021-07-14 07:56:55 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210714-074948-0ed9ca.aar) |
| | | | | | | |
| :heavy_check_mark: | 7fdca71b | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/16/) | 2021-07-14 08:19:17 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210714-081708-7fdca7.zip) |
| :heavy_check_mark: | 7fdca71b | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/16/) | 2021-07-14 08:19:17 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210714-081708-7fdca7.zip) |
| :heavy_check_mark: | 7fdca71b | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/16/) | 2021-07-14 08:21:41 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210714-081703-7fdca7.aar) |
| | | | | | | |
| :heavy_check_mark: | 270326d4 | [#17](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/17/) | 2021-07-14 09:20:25 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210714-091825-270326.zip) |
| :heavy_check_mark: | 270326d4 | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/17/) | 2021-07-14 09:20:47 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210714-091825-270326.zip) |
| :heavy_check_mark: | 270326d4 | [#17](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/17/) | 2021-07-14 09:23:34 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210714-091820-270326.aar) |
| | | | | | | |
| :heavy_check_mark: | 46b46b35 | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/18/) | 2021-07-14 09:40:48 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210714-093811-46b46b.zip) |
| :heavy_check_mark: | 46b46b35 | [#18](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/18/) | 2021-07-14 09:41:29 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210714-093806-46b46b.zip) |
| :heavy_check_mark: | 46b46b35 | [#18](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/18/) | 2021-07-14 09:44:44 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210714-093806-46b46b.aar) |
| | | | | | | |
| :heavy_check_mark: | ccfc7d47 | [#19](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/19/) | 2021-07-14 11:07:45 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210714-110546-ccfc7d.zip) |
| :heavy_check_mark: | ccfc7d47 | [#19](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/19/) | 2021-07-14 11:07:48 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210714-110546-ccfc7d.zip) |
| :heavy_check_mark: | ccfc7d47 | [#19](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/19/) | 2021-07-14 11:12:39 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210714-110546-ccfc7d.aar) |
| | | | | | | |
| :heavy_check_mark: | c5497098 | [#20](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/20/) | 2021-07-14 12:52:10 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210714-124922-c54970.zip) |
| :heavy_check_mark: | c5497098 | [#20](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/20/) | 2021-07-14 12:52:43 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210714-124922-c54970.zip) |
| :heavy_check_mark: | c5497098 | [#20](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/20/) | 2021-07-14 12:53:58 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210714-124922-c54970.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 7834d0a8 | [#21](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/21/) | 2021-07-16 18:11:23 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210716-180853-7834d0.zip) |
| :heavy_check_mark: | 7834d0a8 | [#21](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/21/) | 2021-07-16 18:11:44 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210716-180853-7834d0.zip) |
| :heavy_check_mark: | 7834d0a8 | [#21](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/21/) | 2021-07-16 18:14:52 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210716-180853-7834d0.aar) |
| | | | | | | |
| :heavy_check_mark: | 11029498 | [#22](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2275/22/) | 2021-07-20 08:52:11 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-210720-084949-110294.zip) |
| :heavy_check_mark: | 11029498 | [#22](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2275/22/) | 2021-07-20 08:52:13 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-210720-084949-110294.zip) |
| :heavy_check_mark: | 11029498 | [#22](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2275/22/) | 2021-07-20 08:55:11 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-210720-084949-110294.aar) |

cammellos committed:
Fix some tests and restore private/public RPC client

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

cammellos committed:
Fix some tests & upgrade contracts

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

cammellos committed:
Remove subs test

The test was relying on a method `UnlockAccount` which is not exported
anymore.
This method seems not something we should generally expose, and in any
case it does not seem to be used by the frontend.

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

cammellos committed:
linting and remove unused code

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

cammellos committed:
fix topic pool tests

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

cammellos committed:
Remove metrics test

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

cammellos committed:
Remove e2e tests from steps

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

cammellos committed:
Send dynamic fee tx parameters

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

cammellos committed:
Use marshal binary

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

cammellos committed:
bump version

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty

Samyoul commented on node/rpc.go:
```diff
@@ -0,0 +1,30 @@
+)
+
+// formatName converts to first character of name to lowercase.
+func formatName(name string) string {
```
I think that this function could have a more descriptive name, though I personally appreciate the mystery of `formatName`

```suggestion
func firstCharToLower(name string) string {
```

Samyoul commented on peers/cotopicpool.go:
```diff
@@ -65,7 +66,14 @@ func (t *cacheOnlyTopicPool) ConfirmAdded(server *p2p.Server, nodeID enode.ID) {
 	// If a peer was trusted, it was moved to connectedPeers,
 	// signal was sent and we can safely remove it.
 	if peer, ok := t.connectedPeers[nodeID]; ok {
-		t.removeServerPeer(server, peer)
+		// NOTE: removeServerPeer removes the server peer immediately.
+		// which means the next discovery.summary is not going to include
+		// the peer.
+		// We leave some time so that we ensure the signal is propagated
+		go func() {
+			time.Sleep(200)
+			t.removeServerPeer(server, peer)
+		}()
```
🐱‍💻 Hacker cat approves of your hacks

Samyoul reviewed (approved):
Looks really good. Great job.

ghost commented:
`vendor/github.com/tyler-smith/go-bip39/Gopkg.lock` and `vendor/gopkg.in/olebedev/go-duktape.v3/Gopkg.lock` changed. Pinging @adambabik and @corpetty
